cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewJobQueue.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.preview;
+
+import javax.annotation.Nullable;
+
+/**
+ * Interface designed for holding {@link PreviewJob}
+ */
+public interface PreviewJobQueue {
+  /**
+   * Poll the next available job in the queue.
+   * @param runnerId id of the preview runner which is polling the job
+   * @return {@code PreviewJob} if such job is available in the queue, {@code null} otherwise
+   */
+  @Nullable
+  PreviewJob poll(String runnerId);
```

Comment:
Returns `Optional<PreviewJob>` instead of null

Reply:
How the runnerId is being used? And what format it is? If it is needed, should it be a structured class instead of a String.